Remotion/Data/DomainObjects/Queries/AppContextProvider.cs

```diff
@@ -0,0 +1,14 @@
+using System;
+
+namespace Remotion.Data.DomainObjects.Queries
+{
+  /// <summary>
+  /// Represents the default implementation of the <see cref="IAppContextProvider"/> interface.
+  /// </summary>
+  public class AppContextProvider : IAppContextProvider
+  {
+    public string BaseDirectory => AppContext.BaseDirectory;
+
+    public string RelativeSearchPath => AppDomain.CurrentDomain.RelativeSearchPath;
```

Comment:
We need different implementations for L12 for full framework and .NET 5.

In the case of .NET 5, `null`, `""` or `"."`.
Have a look, without strong consideration of implications on the functionalities, at semantics entailed by the different options.

Interface + Implementation need to move to `Remotion/Core/Core/Utilities`.